components/switch/switch-mixin.js

```diff
@@ -40,20 +41,23 @@ export const SwitchMixin = superclass => class extends RtlMixin(superclass) {
 				padding: 0.1rem;
 				vertical-align: middle;
 			}
-			.d2l-switch-container:focus,
-			.d2l-switch-container:hover {
+			.d2l-switch-container.focus-visible {
 				border-color: var(--d2l-color-celestine);
 			}
+			.d2l-switch-inner:hover {
+				box-shadow: 0 0 0 1px var(--d2l-color-celestine) inset;
+			}
 			:host([disabled]) .d2l-switch-container {
 				cursor: default;
 				opacity: 0.5;
 			}
 			:host([disabled]) .d2l-switch-container:focus,
-			:host([disabled]) .d2l-switch-container:hover {
-				border-color: transparent;
+			:host([disabled]) .d2l-switch-container:hover,
```

Comment:
I think the first line (`:host([disabled]) .d2l-switch-container:hover,`) can be excluded since it doesn't have a `box-shadow`?

Reply:
So I ended up noticing that the hover effect should be applied to the inner div when the container div is being hovered (that is the div listening for the click event) so the class changed slightly but is still necessary. I was able to remove the :focus line as the switch is not selectable via the keyboard anymore so we don't have to worry about the classes being applied